Lib/test/test_threading.py

```diff
@@ -1827,6 +1827,39 @@ def test__all__(self):
         support.check__all__(self, threading, ('threading', '_thread'),
                              extra=extra, not_exported=not_exported)
 
+    @requires_subprocess()
+    def test_gh112826_missing__thread__is_main_interpreter(self):
+        with os_helper.temp_dir() as tempdir:
+            modname = '_thread_fake'
+            import os.path
+            filename = os.path.join(tempdir, modname + '.py')
+            with open(filename, 'w') as outfile:
+                outfile.write("""if True:
+                    import _thread
+                    ns = globals().update(vars(_thread))
+                    #from _thread import *
```

Comment:
What's with the commented out line? Was that import not working? Why mess with globals() at all?

Reply:
> What's with the commented out line? Was that import not working?

Yeah, it wasn't and I wanted to prioritize the fix over taking the time to figure it out. 😄  I hadn't meant to leave the commented-out line though.

> Why mess with globals() at all?

I just wanted to quickly duplicate `_thread` without `_is_main_interpreter()`.  Doing it this way seemed easiest.  (I do need to drop the `ns = ` from line 1839 though.)